{
    "active_lock_reason": null,
    "assignee": {
        "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
        "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
        "followers_url": "https://api.github.com/users/weiji14/followers",
        "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
        "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/weiji14",
        "id": 23487320,
        "login": "weiji14",
        "node_id": "MDQ6VXNlcjIzNDg3MzIw",
        "organizations_url": "https://api.github.com/users/weiji14/orgs",
        "received_events_url": "https://api.github.com/users/weiji14/received_events",
        "repos_url": "https://api.github.com/users/weiji14/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/weiji14"
    },
    "assignees": [
        {
            "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
            "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
            "followers_url": "https://api.github.com/users/weiji14/followers",
            "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
            "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/weiji14",
            "id": 23487320,
            "login": "weiji14",
            "node_id": "MDQ6VXNlcjIzNDg3MzIw",
            "organizations_url": "https://api.github.com/users/weiji14/orgs",
            "received_events_url": "https://api.github.com/users/weiji14/received_events",
            "repos_url": "https://api.github.com/users/weiji14/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/weiji14"
        }
    ],
    "author_association": "MEMBER",
    "body": "**Description of the problem**\r\n\r\nJust noticed that datetime columns being passed into `pygmt.info` doesn't work. This follows on from the `pandas.DataFrame` inputs into `pygmt.info` functionality added in #574, see also #464 and #562 where the datetime machinery should be more or less implemented.\r\n\r\n<!-- Please be as detailed as you can when describing an issue. The more information we have, the easier it will be for us to track this down. -->\r\n\r\n**Full code that generated the error**\r\n\r\n```python\r\nimport pygmt\r\nimport pandas as pd\r\n\r\ntable = pd.DataFrame(data=[1,3,2,5,4], columns=[\"z\"])\r\ntable[\"time\"] = pd.date_range(start=\"2020-01-01\", periods=5)\r\n\r\npygmt.info(table=table)\r\n```\r\n\r\nNote that the equivalent `gmt` command does work on datetime inputs.\r\n\r\n```\r\n!gmt info temp.txt\r\ntemp.txt: N = 5\t<1/5>\t<2020-01-01T00:00:00/2020-01-05T00:00:00>\r\n```\r\n\r\n**Full error message**\r\n\r\n```\r\n---------------------------------------------------------------------------\r\nGMTCLibError                              Traceback (most recent call last)\r\n<ipython-input-6-a9e68c3dc07e> in <module>\r\n----> 1 pygmt.info(table=table)\r\n\r\n~/pygmt/pygmt/helpers/decorators.py in new_module(*args, **kwargs)\r\n    235                 if alias in kwargs:\r\n    236                     kwargs[arg] = kwargs.pop(alias)\r\n--> 237             return module_func(*args, **kwargs)\r\n    238 \r\n    239         new_module.aliases = aliases\r\n\r\n~/pygmt/pygmt/modules.py in info(table, **kwargs)\r\n    116 \r\n    117         with GMTTempFile() as tmpfile:\r\n--> 118             with file_context as fname:\r\n    119                 arg_str = \" \".join(\r\n    120                     [fname, build_arg_string(kwargs), \"->\" + tmpfile.name]\r\n\r\n~/miniconda3/envs/pygmt/lib/python3.7/contextlib.py in __enter__(self)\r\n    110         del self.args, self.kwds, self.func\r\n    111         try:\r\n--> 112             return next(self.gen)\r\n    113         except StopIteration:\r\n    114             raise RuntimeError(\"generator didn't yield\") from None\r\n\r\n~/pygmt/pygmt/clib/session.py in virtualfile_from_matrix(self, matrix)\r\n   1268         )\r\n   1269 \r\n-> 1270         self.put_matrix(dataset, matrix)\r\n   1271 \r\n   1272         with self.open_virtual_file(\r\n\r\n~/pygmt/pygmt/clib/session.py in put_matrix(self, dataset, matrix, pad)\r\n    906         )\r\n    907         if status != 0:\r\n--> 908             raise GMTCLibError(\"Failed to put matrix of type {}.\".format(matrix.dtype))\r\n    909 \r\n    910     def write_data(self, family, geometry, mode, wesn, output, data):\r\n\r\nGMTCLibError: Failed to put matrix of type object.\r\n```\r\n\r\n**System information**\r\n\r\nPlease paste the output of `python -c \"import pygmt; pygmt.show_versions()\"`:\r\n\r\n```\r\nPyGMT information:\r\n  version: v0.1.2+55.g6deb388\r\nSystem information:\r\n  python: 3.7.8 | packaged by conda-forge | (default, Jul 31 2020, 02:25:08)  [GCC 7.5.0]\r\n  executable: ~/miniconda3/envs/pygmt/bin/python\r\n  machine: Linux-4.19.0-8-amd64-x86_64-with-debian-10.5\r\nDependency information:\r\n  numpy: 1.19.1\r\n  pandas: 1.1.1\r\n  xarray: 0.16.0\r\n  netCDF4: 1.5.3\r\n  packaging: 20.4\r\n  ghostscript: 9.27\r\n  gmt: None\r\nGMT library information:\r\n  binary dir: ~/miniconda3/envs/pygmt/bin\r\n  cores: 2\r\n  grid layout: rows\r\n  library path: ~/miniconda3/envs/pygmt/lib/libgmt.so\r\n  padding: 2\r\n  plugin dir: ~/miniconda3/envs/pygmt/lib/gmt/plugins\r\n  share dir: ~/miniconda3/envs/pygmt/share/gmt\r\n  version: 6.1.1\r\n```\r\n",
    "closed_at": "2021-04-22T23:51:46Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
        "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
        "followers_url": "https://api.github.com/users/weiji14/followers",
        "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
        "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/weiji14",
        "id": 23487320,
        "login": "weiji14",
        "node_id": "MDQ6VXNlcjIzNDg3MzIw",
        "organizations_url": "https://api.github.com/users/weiji14/orgs",
        "received_events_url": "https://api.github.com/users/weiji14/received_events",
        "repos_url": "https://api.github.com/users/weiji14/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/weiji14"
    },
    "comments": 9,
    "comments_content": [
        {
            "author_association": "MEMBER",
            "body": "[Here](https://github.com/GenericMappingTools/gmt/blob/d3d64d0027263892efd88e9b23c4d40e167a2a98/src/gmt_api.c#L13374) is the definition of the `GMT_Put_Matrix` function:\r\n```c\r\nint GMT_Put_Matrix (void *API, struct GMT_MATRIX *M, unsigned int type, int pad, void *matrix) \r\n```\r\n\r\nThe third parameter *type* is the data type of the matrix, e.g., `GMT_DOUBLE`, `GMT_FLOAT`. It also means that all elements of the matrix must have the exact same data type. Thus, in PyGMT, we can't pass 2D numpy arrays with mixed data types to `put_matrix` function. \r\n\r\nThe fix seems easy. We may have to pass 2D arrays as a series of vectors, via `virtualfile_from_vectors`.\r\n\r\n",
            "created_at": "2020-09-15T08:02:59Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-692541915",
            "id": 692541915,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDY5MjU0MTkxNQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/692541915/reactions"
            },
            "updated_at": "2020-09-15T08:02:59Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/692541915",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Ping @weiji14. ",
            "created_at": "2020-09-19T17:47:14Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-695337243",
            "id": 695337243,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDY5NTMzNzI0Mw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/695337243/reactions"
            },
            "updated_at": "2020-09-19T17:47:14Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/695337243",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> The third parameter _type_ is the data type of the matrix, e.g., `GMT_DOUBLE`, `GMT_FLOAT`. It also means that all elements of the matrix must have the exact same data type. Thus, in PyGMT, we can't pass 2D numpy arrays with mixed data types to `put_matrix` function.\r\n>\r\n> The fix seems easy. We may have to pass 2D arrays as a series of vectors, via `virtualfile_from_vectors`.\r\n\r\nRight, so we'll need to have something like an if-then or try-except to handle mixed dtypes. A couple of other details to consider:\r\n\r\n1. Do we switch to using `put_vectors` for `info` all the time (will involve a for-loop), or do we check if dtypes are mixed, then use `put_vectors`, else use `put_matrix` as per usual.\r\n\r\nNote that `numpy.array`s always have the same dtype, it will just be `np.object` if dtypes are mixed. `pandas.DataFrame`s are the ones that can explicitly have different dtypes in different columns.\r\n\r\n2. Should we generalize `info` to handle/support other mixed dtype combinations (e.g. int32/float32/etc) properly, thinking about #547 here.\r\n\r\nI've got a unit test for this written up already and will submit a PR soon, just need to work out these implementation details :smile:.",
            "created_at": "2020-09-19T20:59:38Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-695355497",
            "id": 695355497,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDY5NTM1NTQ5Nw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/695355497/reactions"
            },
            "updated_at": "2020-09-19T21:22:25Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/695355497",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Just following up on this, we've merged in #619 so if you install PyGMT from the master branch, passing in `datetime` inputs won't result in \"GMTCLibError: Failed to put matrix of type object.\" anymore. However, the datetime column's ranges will be reported in UNIX timestamps instead of ISO datetimes.\r\n\r\nA **workaround** for this as mentioned at https://github.com/GenericMappingTools/gmt/issues/4241#issuecomment-695958278 is to use something like `pygmt.info(table=df, f=\"1T\")`, which would explicitly tell GMT that the second column is a datetime type, and should be handled that way.\r\n\r\nWe will close this issue once this upstream GMT issue at https://github.com/GenericMappingTools/gmt/issues/4241 is resolved, and perhaps when PyGMT bumps the minimum required version to GMT 6.2.0 and/or when conda GMT 6.2.0.dev builds are available with https://github.com/conda-forge/gmt-feedstock/pull/100.",
            "created_at": "2020-09-22T21:38:59Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-696996503",
            "id": 696996503,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDY5Njk5NjUwMw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/696996503/reactions"
            },
            "updated_at": "2020-09-22T21:38:59Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/696996503",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> A **workaround** for this as mentioned at [GenericMappingTools/gmt#4241 (comment)](https://github.com/GenericMappingTools/gmt/issues/4241#issuecomment-695958278) is to use something like `pygmt.info(table=df, f=\"1T\")`, which would explicitly tell GMT that the second column is a datetime type, and should be handled that way.\r\n\r\nSo the workaround doesn't quite work because of the way we've implemented things in #619 using [`np.loadtxt`](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html):\r\n\r\n```python\r\nimport pandas as pd\r\nimport pygmt\r\n\r\ntable = pd.date_range(start=\"2010-01-01\", end=\"2020-01-01\")\r\npygmt.info(table=table, spacing=\"1Y\", f=\"0T\")\r\n```\r\n\r\nerrors with:\r\n\r\n```python-traceback\r\n---------------------------------------------------------------------------\r\nValueError                                Traceback (most recent call last)\r\n<ipython-input-88-cac984c8d7d8> in <module>\r\n----> 1 pygmt.info(table=df[[time_var, elev_var]], spacing=f\"1W/{spacing}\", f=\"0T\")\r\n\r\n~/miniconda3/envs/pygmt/src/pygmt/pygmt/helpers/decorators.py in new_module(*args, **kwargs)\r\n    268                 if alias in kwargs:\r\n    269                     kwargs[arg] = kwargs.pop(alias)\r\n--> 270             return module_func(*args, **kwargs)\r\n    271 \r\n    272         new_module.aliases = aliases\r\n\r\n~/miniconda3/envs/pygmt/src/pygmt/pygmt/modules.py in info(table, **kwargs)\r\n    137             if result.startswith((\"-R\", \"-T\")):  # e.g. -R0/1/2/3 or -T0/9/1\r\n    138                 result = result[2:].replace(\"/\", \" \")\r\n--> 139             result = np.loadtxt(result.splitlines())\r\n    140 \r\n    141         return result\r\n\r\n~/miniconda3/envs/pygmt/lib/python3.8/site-packages/numpy/lib/npyio.py in loadtxt(fname, dtype, comments, delimiter, converters, skiprows, usecols, unpack, ndmin, encoding, max_rows)\r\n   1137         # converting the data\r\n   1138         X = None\r\n-> 1139         for x in read_data(_loadtxt_chunksize):\r\n   1140             if X is None:\r\n   1141                 X = np.array(x, dtype)\r\n\r\n~/miniconda3/envs/pygmt/lib/python3.8/site-packages/numpy/lib/npyio.py in read_data(chunk_size)\r\n   1065 \r\n   1066             # Convert each value according to its column and store\r\n-> 1067             items = [conv(val) for (conv, val) in zip(converters, vals)]\r\n   1068 \r\n   1069             # Then pack it according to the dtype's nesting\r\n\r\n~/miniconda3/envs/pygmt/lib/python3.8/site-packages/numpy/lib/npyio.py in <listcomp>(.0)\r\n   1065 \r\n   1066             # Convert each value according to its column and store\r\n-> 1067             items = [conv(val) for (conv, val) in zip(converters, vals)]\r\n   1068 \r\n   1069             # Then pack it according to the dtype's nesting\r\n\r\n~/miniconda3/envs/pygmt/lib/python3.8/site-packages/numpy/lib/npyio.py in floatconv(x)\r\n    761         if '0x' in x:\r\n    762             return float.fromhex(x)\r\n--> 763         return float(x)\r\n    764 \r\n    765     typ = dtype.type\r\n\r\nValueError: could not convert string to float: '2019-05-19T20:53:51'\r\n```\r\n\r\n[np.loadtxt](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html) assumes that the text are to be read as floating point numbers, but datetimes like \"2019-05-19T20:53:51\" are not floats. We'll need to set the dtype using `np.loadtxt(..., dtype=???)`, where `???` is `\"str,float\"` or something (ref https://stackoverflow.com/a/31554777/6611055).\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/c7c5eaecf8fedbca584743f6adca4a378851ba9a/pygmt/modules.py#L139",
            "created_at": "2020-11-11T02:38:52Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-725091467",
            "id": 725091467,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDcyNTA5MTQ2Nw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/725091467/reactions"
            },
            "updated_at": "2020-11-11T04:38:15Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/725091467",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Alright, with #960 merged. Anyone installing PyGMT from the master branch (see https://www.pygmt.org/v0.3.0/install.html#using-pip) should be able to use the `coltypes=\"0T\"` GMT 6.1.1 workaround (where `0T` means the first column contains time), i.e.:\r\n\r\n```python\r\nimport pandas as pd\r\nimport pygmt\r\n\r\ntable = pd.date_range(start=\"2010-01-01\", end=\"2020-01-01\")\r\nregion = pygmt.info(table=table, spacing=\"1Y\", coltypes=\"0T\")\r\nprint(region)\r\n# ['2010-01-01T00:00:00' '2020-01-01T00:00:00' '0' '0']\r\n```\r\nAssuming that https://github.com/GenericMappingTools/gmt/issues/4241 is resolved in GMT 6.2.0, then GMT 6.2.0 users won't need to use the `coltypes` parameter in the future (saves people from needing to know what is the number of the time column).",
            "created_at": "2021-03-03T22:26:18Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-790117248",
            "id": 790117248,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc5MDExNzI0OA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/790117248/reactions"
            },
            "updated_at": "2021-03-03T22:26:18Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/790117248",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "FYI, https://github.com/GenericMappingTools/gmt/issues/4241 has been magically resolved, so this issue can be resolved when PyGMT bumps the minimum version to GMT 6.2.0!",
            "created_at": "2021-03-14T21:01:07Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-798979009",
            "id": 798979009,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc5ODk3OTAwOQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 1,
                "rocket": 0,
                "total_count": 2,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/798979009/reactions"
            },
            "updated_at": "2021-03-14T21:01:07Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/798979009",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Fixed by https://github.com/GenericMappingTools/gmt/pull/4849",
            "created_at": "2021-03-14T21:09:49Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-798980292",
            "id": 798980292,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc5ODk4MDI5Mg==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/798980292/reactions"
            },
            "updated_at": "2021-03-14T21:09:49Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/798980292",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
                "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
                "followers_url": "https://api.github.com/users/maxrjones/followers",
                "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
                "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/maxrjones",
                "id": 14077947,
                "login": "maxrjones",
                "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
                "organizations_url": "https://api.github.com/users/maxrjones/orgs",
                "received_events_url": "https://api.github.com/users/maxrjones/received_events",
                "repos_url": "https://api.github.com/users/maxrjones/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/maxrjones"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Phew, thanks team, glad to close down another >6 month old issue!",
            "created_at": "2021-04-22T23:53:35Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-825273663",
            "id": 825273663,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgyNTI3MzY2Mw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/825273663/reactions"
            },
            "updated_at": "2021-04-22T23:53:35Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/825273663",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597/comments",
    "created_at": "2020-09-10T10:01:12Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/597",
    "id": 697707713,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        },
        {
            "color": "e8ff6b",
            "default": false,
            "description": "Bug or missing feature of upstream core GMT",
            "id": 2059525605,
            "name": "upstream",
            "node_id": "MDU6TGFiZWwyMDU5NTI1NjA1",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/upstream"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": "2021-06-21T14:49:45Z",
        "closed_issues": 244,
        "created_at": "2020-11-21T04:47:19Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
            "events_url": "https://api.github.com/users/seisman/events{/privacy}",
            "followers_url": "https://api.github.com/users/seisman/followers",
            "following_url": "https://api.github.com/users/seisman/following{/other_user}",
            "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/seisman",
            "id": 3974108,
            "login": "seisman",
            "node_id": "MDQ6VXNlcjM5NzQxMDg=",
            "organizations_url": "https://api.github.com/users/seisman/orgs",
            "received_events_url": "https://api.github.com/users/seisman/received_events",
            "repos_url": "https://api.github.com/users/seisman/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/seisman"
        },
        "description": "",
        "due_on": "2021-06-20T07:00:00Z",
        "html_url": "https://github.com/GenericMappingTools/pygmt/milestone/6",
        "id": 6133205,
        "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/6/labels",
        "node_id": "MDk6TWlsZXN0b25lNjEzMzIwNQ==",
        "number": 6,
        "open_issues": 0,
        "state": "closed",
        "title": "0.4.0",
        "updated_at": "2021-11-03T23:29:33Z",
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/6"
    },
    "node_id": "MDU6SXNzdWU2OTc3MDc3MTM=",
    "number": 597,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597/timeline",
    "title": "Running info on pandas.DataFrame with time column doesn't work",
    "updated_at": "2021-04-22T23:53:35Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/597",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
        "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
        "followers_url": "https://api.github.com/users/weiji14/followers",
        "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
        "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/weiji14",
        "id": 23487320,
        "login": "weiji14",
        "node_id": "MDQ6VXNlcjIzNDg3MzIw",
        "organizations_url": "https://api.github.com/users/weiji14/orgs",
        "received_events_url": "https://api.github.com/users/weiji14/received_events",
        "repos_url": "https://api.github.com/users/weiji14/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/weiji14"
    }
}